### PR TITLE
cc-check: Isolate architecture specific code

### LINKED ISSUE #992: Rework cc-check to isolate architecture specific code
Currently, the `cc-check` command assumes an `amd64` system. Rework to make the code easier to support other architectures.

### PR DESCRIPTION
**Pull request by jodh-intel**
Reworked the `cc-check` command to make it easier to support other
architectures.

Fixes #992.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Due to all the code shuffling, this is going to be rather confusing to review I suspect.

Blocked on #995.

**Comment by grahamwhaley** [comment]
X-)
I'll give this a visual run/check once #995 has landed then...

File: cc-check.go [hunk 1]
```diff
@@ -45,6 +37,13 @@ type kernelModule struct {
 	parameters map[string]string
 }
 
+type vmContainerCapableDetails struct {
+	cpuinfoFile           string
```
Comment (sboeuf) [inline cc-check.go]: s/cpuinfoFile/cpuInfoFile/

File: cc-env.go [hunk 2]
```diff
@@ -175,7 +175,15 @@ func getHostInfo() (HostInfo, error) {
...
-	err = hostIsClearContainersCapable(procCPUInfo)
+
+	details := vmContainerCapableDetails{
+		cpuinfoFile:           procCPUInfo,
+		requiredCPUFlags:      archRequiredCPUFlags,
+		requiredCPUAttribs:    archRequiredCPUAttribs,
+		requiredKernelModules: archRequiredKernelModules,
+	}
+
+	err = hostIsVMContainerCapable(details)
```
Comment (sboeuf) [inline cc-env.go]:
```go
if err := hostIsVMContainerCapable(details); err != nil {
```
would be better

**Comment by sboeuf** [comment]
@jodh-intel thanks for this change, this makes things architecture agnostic, that's great :+1:

File: cc-check.go (hunk 1 above)
Comment (jodh-intel) [reply cc-check.go]: Fixed.

File: cc-env.go (hunk 2 above)
Comment (jodh-intel) [reply cc-env.go]: Fixed.

**Comment by jodh-intel** [comment]
Thanks @sboeuf - branch updated.

**Comment by sboeuf** [comment]
LGTM

**Comment by jodh-intel** [comment]
Hi @grahamwhaley - now that #993 has landed, could you take a look at this?

**Comment by jodh-intel** [comment]
Hi @devimc - could you take a look?

**Comment by grahamwhaley** [comment]
Ignoring the metrics CI fail - the machine had a glitch that we are in the process of fixing.

**Pull request merged**